src/protocol/parser.rs

```diff
@@ -99,32 +99,36 @@ impl Parser {
         })
     }
 
-    pub fn parse_command(&mut self) -> R<Command> {
-        let executable = match self.parse_word()? {
-            None => self.parse_error("expected: space-separated command and arguments")?,
+    pub fn parse_command(command: &str) -> R<Command> {
```

Comment:
I wonder if it would be good to move the creation of `Command` out into where this is called. So that this module only deals with argument parsing and nothing else. Then we'd have a nice module with a tiny interface that's just `parse_arguments(string: &str) -> Vec<String>`.